engine/execution/ingestion/engine.go

```diff
@@ -141,6 +149,12 @@ func New(
 // successfully started.
 func (e *Engine) Ready() <-chan struct{} {
 	if !e.pauseExecution {
+		if uploadEnabled {
+			if err := e.computationManager.RetryUpload(); err != nil {
```

Comment:
Is there a reason to keep `RetryUpload` as a part of `ComputationManager`? It doesn't seem to be related, and at this point `RetryUpload` doesn't really have anything to do with computation

Reply:
the reason is all uploaders are being kept in `ComputationManager`. I will need to access all uploaders to make this call.

Reply:
Hmm, I don't think thats the best approach. ie. `ComputationManager` doesn't do retry, it can find (or not!) some uploaders and call retry.
Would be better to store a pointer to retriableUploader in the engine (or some other place?) and make a call explicitly, not involving Manager at all.

Reply:
addressed. now keeping uploader references in engine. the `RetryUpload()` call now will not involve `ComputationManager`